polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -646,6 +671,41 @@ enum MessageSource {
 	Local,
 }
 
+// Encountered error while validating an assignment.
+#[derive(Debug)]
+enum InvalidAssignmentError {
+	// The vrf check for the assignment failed.
+	#[allow(dead_code)]
```

Comment:
why is this annotation needed here and below?

Reply:
Because of this: https://github.com/rust-lang/rust/issues/123068, compiler says this filed is never used, although it is used when printing the error.